src/main/java/cloud/eppo/ufc/dto/OperatorType.java

```diff
@@ -14,20 +17,40 @@ public enum OperatorType {
   IS_NULL("IS_NULL");
 
   public final String value;
+  private static final Map<String, OperatorType> valuesToOperatorType =
+      buildValueToOperatorTypeMap();
+  private static final Map<String, OperatorType> hashesToOperatorType =
+      buildHashToOperatorTypeMap();
+
+  private static Map<String, OperatorType> buildValueToOperatorTypeMap() {
+    Map<String, OperatorType> result = new HashMap<>();
+    for (OperatorType type : OperatorType.values()) {
+      result.put(type.value, type);
+    }
+    return result;
+  }
+
+  private static Map<String, OperatorType> buildHashToOperatorTypeMap() {
+    Map<String, OperatorType> result = new HashMap<>();
+    for (OperatorType type : OperatorType.values()) {
+      result.put(getMD5Hex(type.value), type);
+    }
+    return result;
+  }
 
   OperatorType(String value) {
     this.value = value;
   }
 
   public static OperatorType fromString(String value) {
-    for (OperatorType type : OperatorType.values()) {
-      if (type.value.equals(value)
-          || getMD5Hex(type.value).equals(value)
-          || getMD5Hex(type.value).equals(value)) {
-        return type;
-      }
+    // First we try obfuscated lookup as in client situations we'll care more about ingestion
+    // performance
+    OperatorType type = hashesToOperatorType.get(value);
```

Comment:
We really shouldn't be hashing operators more than once so as a performance improvement we remove the loop and just built two maps for unhashed and hashed operators on init.